### DISCUSSION TITLE
Difference between Requeue vs. RequeueAfter?

### DISCUSSION THREAD

**Discussion by alex067** [kubernetes-sigs/kubebuilder#3299] [Support] [answered]
Hey everyone.

I'm new to Kubebuilder and had a quick question regarding Requeue. 

I believe the operator by default, reconciles every 10 hours (aside from reacting to events). If I wanted to requeue a reconciled request, and simply use: {Requeue: true}, does that infer that in 10 hours we're going to try to reconcile again?

Or is there a separate time that this runs on?

I understand that RequeueAfter lets you specify a specific time, but what time unit is this considering? Milli seconds?

Thank you all!

**Comment by camilamacedo86** [comment, marked as answer]
Hi @alex067,

> I believe the operator by default, reconciles every 10 hours (aside from reacting to events).

Yes, that is true. It is the default time define in the Manager Options. See: 
- https://github.com/kubernetes-sigs/controller-runtime/blob/v0.16.2/pkg/cache/cache.go#L135-L160
- https://github.com/kubernetes-sigs/controller-runtime/issues/521

>  If I wanted to requeue a reconciled request, and simply use: {Requeue: true}, does that infer that in 10 hours we're going to try to reconcile again?

No, the {Requeue: true} will cause the reconcile loop to start from the beginning immediately. It's useful, for instance, when you've updated a resource and want to re-fetch it to ensure all operations are executed correctly. Remember, it's important to implement idempotent solutions.

I'd recommed you check the content of this PR: https://github.com/kubernetes-sigs/kubebuilder/blob/a1c6eb502aa5c626d8ce7d676f07211521cc61d8/docs/book/src/getting-started.md . It will clarify a lot how things work,

Closing this one as answered.
However, if you need feel free to re-open,
